tools/src/test/java/org/neo4j/tools/org/neo4j/index/GBPTreePlayground.java

```diff
@@ -0,0 +1,233 @@
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.tools.org.neo4j.index;
+
+import org.apache.commons.lang3.mutable.MutableBoolean;
+import org.apache.commons.lang3.mutable.MutableLong;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.PrintStream;
+
+import org.neo4j.index.internal.gbptree.GBPTree;
+import org.neo4j.index.internal.gbptree.RecoveryCleanupWorkCollector;
+import org.neo4j.index.internal.gbptree.SimpleLongLayout;
+import org.neo4j.index.internal.gbptree.Writer;
+import org.neo4j.io.fs.DefaultFileSystemAbstraction;
+import org.neo4j.io.fs.FileSystemAbstraction;
+import org.neo4j.io.pagecache.IOLimiter;
+import org.neo4j.io.pagecache.PageCache;
+import org.neo4j.io.pagecache.impl.muninn.StandalonePageCacheFactory;
+import org.neo4j.kernel.lifecycle.LifeSupport;
+import org.neo4j.tools.console.input.Command;
+import org.neo4j.tools.console.input.ConsoleInput;
+
+import static org.neo4j.index.internal.gbptree.GBPTree.NO_HEADER_READER;
+import static org.neo4j.index.internal.gbptree.GBPTree.NO_HEADER_WRITER;
+import static org.neo4j.index.internal.gbptree.GBPTree.NO_MONITOR;
+import static org.neo4j.tools.console.input.ConsoleUtil.staticPrompt;
+
+public class GBPTreePlayground
+{
+    private final File indexFile;
+    private GBPTree<MutableLong,MutableLong> tree;
+
+    private int pageSize = 256;
+    private PageCache pageCache;
+    private SimpleLongLayout layout;
+    private final MutableBoolean autoPrint = new MutableBoolean( true );
+
+    private GBPTreePlayground( FileSystemAbstraction fs, File indexFile )
+    {
+        this.indexFile = indexFile;
+        this.layout = SimpleLongLayout.longLayout().build();
+        this.pageCache = StandalonePageCacheFactory.createPageCache( fs );
+    }
+
+    private void setupIndex() throws IOException
+    {
+        tree = new GBPTree<>( pageCache, indexFile, layout, pageSize, NO_MONITOR, NO_HEADER_READER, NO_HEADER_WRITER,
+                RecoveryCleanupWorkCollector.IMMEDIATE );
+    }
+
+    private void run() throws InterruptedException, IOException
+    {
+        setupIndex();
+
+        LifeSupport life = new LifeSupport();
+        ConsoleInput consoleInput = life.add( new ConsoleInput( System.in, System.out, staticPrompt( "# " ) ) );
+        consoleInput.add( "print", new Print() );
+        consoleInput.add( "add", new AddCommand() );
+        consoleInput.add( "remove", new RemoveCommand() );
+        consoleInput.add( "cp", new Checkpoint() );
+        consoleInput.add( "autoprint", new ToggleAutoPrintCommand() );
+
+        life.start();
+        try
+        {
+            consoleInput.waitFor();
+        }
+        finally
+        {
+            life.shutdown();
+        }
+    }
+
+    private class Print implements Command
+    {
+        @Override
+        public void run( String[] args, PrintStream out ) throws Exception
+        {
+            tree.printTree();
```

Comment:
Could be useful to have an option to print state pages too, especially since there's a checkpoint command too.